functions/all_words.py

```
@@ -571,119 +581,166 @@ def smush_all_plan(results, outer_uses, run_budget=1500, time_limit=1.5):
+        # A letter whose entire word supply can't cover its remaining uses can
+        # never be flattened by this candidate set.
+        coverable = {i for i in unfinished if supply[i] >= start[i]}
+        if exact_only and coverable != unfinished:
+            return None
+
+        # Pangram-first attempts: consume a pangram up front and solve the
+        # rest exactly. Kept only when fully flat, so completeness never
+        # regresses and the clean plate is never traded for the pangram.
+        solved = False
+        for psig in pangram_sigs[:3]:
+            if any(c > n for c, n in zip(psig, start)):
```

Ruff (0.15.21)

[warning] `zip()` without an explicit `strict=` parameter

Add explicit value for parameter `strict=`

(B905)